frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js

```diff
@@ -298,6 +298,34 @@ describe("scenarios > admin > settings", () => {
     });
   }
 
+  it("'General' admin settings should handle setup via `MB_SITE_ULR` environment variable (metabase#14900)", () => {
+    cy.server();
+    // 1. Get the array of ALL available settings
+    cy.request("GET", "/api/setting").then(({ body }) => {
+      // 2. Create a stubbed version of that array by passing modified "site-url" settings
+      const STUBBED_BODY = body.map(setting => {
+        if (setting.key === "site-url") {
+          const STUBBED_SITE_URL = Object.assign({}, setting, {
+            is_env_setting: true,
+            value: null,
+          });
+
+          return STUBBED_SITE_URL;
+        }
+        return setting;
```

Comment:
Was thinking of using ternary operator here but thought that might compromise the legibility?

```javascript
setting.key === "site-url"
  ? Object.assign({}, setting, {
      is_env_setting: true,
      value: null,
    })
  : setting;
```

Reply:
I'm more pro-ternary than most and prefer that here, but I expect most people would disagree. 🤷

Reply:
I don't mind using it. It **always** looks more beautiful and terse to me. I was just wondering how much is lost in the abstraction and will we know at the first glance what's going on here in 6 months :)